mesonbuild/scripts/python_info.py

```diff
@@ -56,6 +60,9 @@ def get_install_paths():
             install_paths = get_distutils_paths(scheme='deb_system', prefix='')
             return paths, install_paths
 
+    if scheme == 'osx_framework_library':
+        scheme = 'posix_prefix'
```

Comment:
This scheme is specific to homebrew itself? I can't tell from the name. The "deb_system" scheme is very obviously specific to deb systems which is why we assume from its presence that we're on debian without anything fancier like checking for an environment variable or for /etc/os-release or something like that.

What are the different values returned by each, when configured in homebrew?

Reply:
ah, good point, it seems that this scheme is also used by the built in python on apple.

https://github.com/Homebrew/homebrew-core/blob/17ec094a27c1535f8644be2120e8bdc8cba20bbd/Formula/p/python%403.12.rb#L75

Probably worth me adding a check for env['HOMEBREW_PREFIX'] as well or instead.

Reply:
hmm, no, i'm unhappy with that approach, will see if there's a better way to be sure we're in homebrew's python

Reply:
maybe the same logic as [pip uses](https://github.com/pypa/pip/blob/9f213bf69ac32c60c84055261c862ff169389e43/src/pip/_internal/locations/_sysconfig.py#L29) would work .

I'll give it a try

Reply:
I am not sure for the need to only do this for homebrew. When using system python you are gonna get a directory that is not writable regardless of what scheme you use. I feel like posix_prefix is acceptable for all other environments on macos

Reply:
just found that meson tests fail on plain OSX, so maybe i'm over concerned .. ':)

Reply:
@eli-schwartz what do you think ti what @kjmeagher says above? I think it would be fine for both osx python and homebrew python if we just use posix_prefix.

Separately I've been seeing if I can extend the macosx CI to cover these variants  - currently it's actually deleting the system python libs for test and using the homebrew python afaict.

Reply:
It's not the system python libs, it's the github actions ones. I do not trust github to get this right. :( Especially since they are literally overwriting files owned by a foreign package manager.

Reply:
I guess I'm having trouble understanding the specific concern - we don't really need to use the setup-python action on the osx builds (see the wip at https://github.com/robtaylor/meson/actions/runs/6540533035 - currently failing because not put in the relevent skips)

Reply:
But I think, having done a fair bit of digging around this, its never really the right thing for us to install with the osx_framework_library scheme, only system installed libraries should be using that. I suspect its actually a bug in the homebrew packaging that its returning this as default_scheme and preffered_scheme, will discuss there. Of course, the macos system python doesn't report these as its python 3.9. 

Will update the patch if this thinking works for you.